cloud/k8s/deploy/cm-kkm-install.yaml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Kontain
+# Derived from:
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kkm-install-entrypoint
+  labels:
+    app: kontain-init
+data:
+  entrypoint.sh: |
+    #!/usr/bin/env bash
+
+    set -x
+    set -euo pipefail
+
+    ROOT_MOUNT_DIR="${ROOT_MOUNT_DIR:-/root}"
+
+    echo "Installing dependencies"
+    apt-get update
+    apt-get install -y make gcc git
```

Comment:
What we discovered in our CI is that these occasionally fail when there is "unattended upgrade" running. In CI we implemented error check and loop. I don't know it this really applies here, let's discuss.

Reply:
Something to think about: Not sure if an update to the OS is a good idea inside a Kubernetes deployment manifest.  Nodes probably needs to be upgraded or updated on a different schedule outside the purview of Kubernetes.

Reply:
This isn't an upgrade. All `apt-get update` does is update the package indexes so `apt-get install` is pulling from fresh places. `apt-get upgrade` is the command that upgrades the software on the local box. We aren't doing that.